### PR TITLE
Add statement_count to aws_iam_policy

### PR DESCRIPTION
**Pull request by tmonk42**
Signed-off-by: Nathan Haneysmith <nathan@chef.io>

### Description

This feature is already in the docs, just adding functionality back in

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [ ] New functionality includes integration tests/controls
- [ ] New Terraform resources
- [ ] Documentation provided or updated for resources 
- [X] All Integration Tests pass
- [X] All Unit Tests pass
- [X] `rake lint` passes
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_iam_policy.rb [hunk 1]
```diff
@@ -85,6 +85,14 @@ def has_statement?(criteria = {})
     statement_match
   end
 
+  def statement_count
+    return false unless @policy_document
+    puts "DEBUG policy doc: #{@policy_document.inspect}"
```
Comment (r-fennell) [inline libraries/aws_iam_policy.rb]: Is this needed?

File: libraries/aws_iam_policy.rb [hunk 2]
```diff
@@ -85,6 +85,14 @@ def has_statement?(criteria = {})
...
+    document = JSON.parse(URI.decode_www_form_component(@policy_document.policy_version.document))
+    statements = document['Statement'].is_a?(Hash) ? [document['Statement']] : document['Statement']
```
Comment (r-fennell) [inline libraries/aws_iam_policy.rb]: Is it possible to have the `Statement` key here a symbol?

File: libraries/aws_iam_policy.rb (hunk 1 above)
Comment (tmonk42) [reply libraries/aws_iam_policy.rb]: Fixed

File: libraries/aws_iam_policy.rb (hunk 2 above)
Comment (tmonk42) [reply libraries/aws_iam_policy.rb]: Hurm, I'll test it. Was following the pattern from existing 'has_statement' section just above.

Comment (tmonk42) [reply libraries/aws_iam_policy.rb]: Updated JSON.parse options to create symbols.

**Pull request merged**